src/nfa/nfa.cc

```diff
@@ -953,12 +953,9 @@ size_t Nfa::get_num_of_trans() const
 {
     size_t num_of_transitions{};
 
-    for (const auto& state_transitions: transition_relation)
+    for (const auto& state_transitions: delta)
```

Comment:
Discussing this with @martinhruska right now. There is an unambiguity with what the call to `delta.size()` should actually return. Up until now, we have considered the result to be the number of states in the whole automaton (and as such it is used on numerous places). From now on, though, if delta is iterable over transitions (if delta returns an iterator iterating over triples `(source_state, symbol. target_state)`), it would make sense that the call to `delta.size()` returns the number of transitions in the automaton. We should discuss this in our meeting and decide which option do we prefer and how to solve the remaining one.